nixos/modules/security/dhparams.nix

```diff
@@ -34,10 +40,19 @@ in
         type = types.str;
         default = "/var/lib/dhparams";
       };
+
+      enable = mkOption {
+        description =
+          ''
+            Whether to generate new DH params and clean up old DH params.
+          '';
+        default = false;
+        type = types.bool;
+      };
     };
   };
 
-  config.systemd.services = {
+  config.systemd.services = if !cfg.enable then {} else {
```

Comment:
Hmm, I thought mkIf cfg.enable pushed the evaluation as far down as possible to avoid looping dependencies, and thus was slower?

Anyway, here is a version using optionalAttrs :)